### PR TITLE
SAK-39989 Sitestats unit test modernization

### PR DESCRIPTION
**Pull request by ern**

### CODE REVIEW

File: sitestats/sitestats-impl/src/test/org/sakaiproject/sitestats/test/SiteStatsTestConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+    public UsageSessionService usageSessionService() {
+        return mock(UsageSessionService.class);
+    }
+
+    @Bean(name = "org.sakaiproject.time.api.UserTimeService")
+    public UserTimeService userTimeService() {
+        return mock(UserTimeService.class);
+    }
+
+}
```
Comment (mpellicer) [inline sitestats/sitestats-impl/src/test/org/sakaiproject/sitestats/test/SiteStatsTestConfiguration.java]: Add an EOF here

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 18
- Added 8
           

Complexity increasing per file
==============================
- sitestats/sitestats-impl/src/test/org/sakaiproject/sitestats/test/SiteStatsTestConfiguration.java  3
         

Complexity decreasing per file
==============================
+ sitestats/sitestats-impl/src/test/org/sakaiproject/sitestats/test/EventAggregatorTestPerf.java  -1
+ sitestats/sitestats-impl/src/test/org/sakaiproject/sitestats/test/StatsUpdateManagerTest.java  -4
+ sitestats/sitestats-impl-hib/src/java/org/sakaiproject/sitestats/impl/SitePresenceTotalImpl.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/sakaiproject/sakai/pullRequest?prid=5150372&bid=16915705)

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 18
- Added 5
           

Complexity increasing per file
==============================
- sitestats/sitestats-impl/src/test/org/sakaiproject/sitestats/test/SiteStatsTestConfiguration.java  3
         

Complexity decreasing per file
==============================
+ sitestats/sitestats-impl/src/test/org/sakaiproject/sitestats/test/StatsUpdateManagerTest.java  -4
+ sitestats/sitestats-impl-hib/src/java/org/sakaiproject/sitestats/impl/SitePresenceTotalImpl.java  -2
+ sitestats/sitestats-impl/src/test/org/sakaiproject/sitestats/test/EventAggregatorTestPerf.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/sakaiproject/sakai/pullRequest?prid=5150372&bid=16915705)

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 19
- Added 4
           

Complexity increasing per file
==============================
- sitestats/sitestats-impl/src/test/org/sakaiproject/sitestats/test/SiteStatsTestConfiguration.java  3
         

Complexity decreasing per file
==============================
+ sitestats/sitestats-impl/src/test/org/sakaiproject/sitestats/test/EventAggregatorTestPerf.java  -1
+ sitestats/sitestats-impl-hib/src/java/org/sakaiproject/sitestats/impl/SitePresenceTotalImpl.java  -2
+ sitestats/sitestats-impl/src/test/org/sakaiproject/sitestats/test/StatsUpdateManagerTest.java  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/sakaiproject/sakai/pullRequest?prid=5150372&bid=16915705)

**Pull request merged**